app/mapper/CaseRequestMapper.scala

```diff
@@ -59,17 +42,17 @@ class CaseRequestMapper {
     val sampleProvided: Option[WhenToSendSample] = answers.get(WhenToSendSamplePage)
     val returnSample: Option[ReturnSamples] = answers.get(ReturnSamplesPage)
 
-    val contact = contactDetails.map(toContact).get
-    val holder: EORIDetails = registeredAddressForEori.map(toHolder(eoriNumber)).get
+    val contact = contactDetails.map(toContact).getOrElse(throwError("contact details"))
 
-    val agentDetails = buildAgentDetails(answers)
+    val agentDetails: Option[AgentDetails] = buildAgentDetails(userEori)(answers)
+    val holderDetails: EORIDetails = buildHolderDetails(userEori)(answers)
 
     val app = Application(
-      holder = holder,
+      holder = holderDetails,
       contact = contact,
       agent = agentDetails,
       offline = false,
-      goodName = describeYourItem.map(_.field1).getOrElse("CaseRequestMapperTestN/A"),
+      goodName = describeYourItem.map(_.field1).getOrElse("N/A"),
       goodDescription = describeYourItem.map(_.field2).getOrElse("N/A"),
```

Comment:
Are we happy to save `N/A`?

Reply:
Maybe we could be more restrictive there by throwing and exception or stopping the process

Reply:
Ultimately this is mandatory data that which is strictly necessary to finish the process

Reply:
Ok. I will throw an error